### ISSUE TITLE
MVC UI: decouple jQuery

### ISSUE THREAD

**Issue by leonkosak** [abpframework/abp#17093]
Due to historical reasons, ASP.NET MVC (released Q4 2007) is almost "tightened" to jQuery (released mid 2006).
Fast forward 15+ years later, pure javascript progressed so much that sometimes jQuery is not needed anymore in some projects.

Would be possible to decouple jQuery at least from LeptonX (and LeptonX-Lite)?
It would also be recommended, that Javascript API (https://docs.abp.io/en/abp/latest/UI/AspNetCore/JavaScript-API/Ajax) would ditch jQuery and implement methods based on modern pure javascript.

Of course, a user could include jQuery in the abp project (but jQuery would not be included when abp project would be generated.

What do you think @hikalkan and @maliming?

**Comment by hikalkan** [comment]
We tought that a few times in the past, however there were library dependencies and also JQuery was still better than pure js. We will re-consider it in 8.0 version, thanks.

**Comment by leonkosak** [comment]
Yes @hikalkan, this would require major development investment and should also be defined as breaking change (because custom user code in MVC abp application usually uses jQuery).

However, if we look DevExtreme components for instance, there are noticeable benefits using non-jquery version: https://community.devexpress.com/blogs/donw/archive/2017/11/09/devextreme-with-and-without-jquery-v17-2.aspx

It is true, that Devextreme MVC wrappers (Nuget libraries) uses jQuery, but if you use plain Devextreme in conjunction with typescript, you can get a comparable "strongly-typed" experience if you would use typescript in abp (https://github.com/abpframework/abp/issues/9005) and abp javascript proxy (https://docs.abp.io/en/abp/latest/UI/AspNetCore/Dynamic-JavaScript-Proxies)

This is just one case, how to obey jQuery without compromising developer or user experience.

I think, that MVC (Razor Pages) will still be a better option compared to Angular and Blazor for building specific applications in many cases for years to come.
Removing jQuery from MVC option would represent modernization of MVC frontend.

**Comment by jasenf** [comment]
I, for one, would prefer to keep jQuery tightly bound to the MVC app.  I appreciate the ease and reduced overhead of not including or needing the more "modern" frameworks (which I consider mostly overkill).

Just my 2-cents.

**Comment by leonkosak** [comment]
Why tightly bounded? You can include jQuery if you prefer using it.
Some applications are not complex on GUI, so why would you include jQuery if you can easy use modern pure javascript for all your operations on frontend (with comparable lines of code)?

**Comment by jasenf** [comment]
I appreciate the way abp currently leverages jquery and how they build on top of it for their MVC apps.  I build on top of it daily.  I have no interest in having it decoupled, there's no upside.   if you want to use a different client framework go for it, but the current "out of box" experience of abp is wonderful.  again, just my opinion.

**Comment by hikalkan** [comment]
> Why tightly bounded? You can include jQuery if you prefer using it.

Designing everything modular and optional and maintaining such a design is not easy. We have used JQuery in the ABP Framework's MVC UI ([see](https://github.com/abpframework/abp/blob/dev/npm/packs/jquery/src/abp.jquery.js)), we have built a lot of module UIs on top of JQuery (we need to re-write all of them to make them working without JQuery). We are using some UI [libraries](https://github.com/abpframework/abp/tree/dev/npm/packs) depends on JQuery in the startup templates and made integrations with these libraries, we have built UI themes that use JQuery, we have create ABP Suite that generates code with JQuery, probably I forgot some of them.

We may make an empty ABP project that doesn't use JQuery, but without all these work, such an empty solution won't be used in practical.

To be honest, it is not easy to drop JQuery from MVC UI, and I think it is not so necessary. There are many JQuery plugins still do not have a good alternative without JQuery. So, we will consider this, however, it is not an easy decision.

**Comment by leonkosak** [comment]
Of course. I have "step-by-step many-years transition" away from jQuery in MVC UI in mind, because I believe that MVC as concept is a more optimal solution (especially for developing long-term support application) as using one of SPA frameworks.
So, reducing "historical" js libraries (especially as big as jQuery) over a long period of time should be considered a good strategy.

**Comment by worthy7** [comment]
HI @hikalkan  I also just hit this. My landing page for the app needs some backend components and Abp "requires" jquery but in reality I see almost no code using javascript (after removing block/spin/toastr). Since my app is Angular Material, I have replaced use of those three anyway with simple busy/blocking directive spinner and snackbar for notifications. I also removed moment.

But the last piece, jquery, which really bloats up my script.js, is only dependent on abp and that's because of abp's initialization code.

I think you could at least just rewrite the small abp code to not use jquery at all, and only have block/messages code use jquery and write that it is a dependency for those two.
It can still be included as base in the template, but then we can easily just delete those files if we don't want them.

**Comment by worthy7** [comment]
This is from Lighthouse for the landing page. There are only 6 js files in my bundle(jquery, signalr, sweetalert, abp, abp jquery, abp.sweet-alert), all of the red is "unused" - although it says 200kb, it's actually only 60kb when i view the network tab.

![Image](https://github.com/user-attachments/assets/ed9fd21a-6a09-49c5-abda-da078f5ba1dd)
